iothub/device/src/Authentication/IotHubConnectionStringBuilder.cs

```diff
@@ -245,8 +221,6 @@ private void Parse(string iotHubConnectionString)
             UsingX509Cert = GetConnectionStringOptionalValueOrDefault<bool>(map, X509CertPropertyName, ParseX509, true)
                 || GetConnectionStringOptionalValueOrDefault<bool>(map, CommonX509CertPropertyName, ParseX509, true);
             GatewayHostName = GetConnectionStringOptionalValue(map, GatewayHostNamePropertyName);
-
-            Validate();
```

Comment:
Removed the implicit call to `Validate()` during `Parse()` because it needed to be called after the object was initialized. Because we assign the auth method (which modifies the object) then parse the connection string in one case, and do it in reverse order in the other case, this meant validation didn't happen when it should.

So now it always happens explicitly (like in the other ctor), which changes the behavior of the object ever so slightly (removed a unit test, changed an expected exception type in another), which I don't think will have much impact.